src/aggregator/aggregator/elem_base.go

```diff
@@ -185,8 +188,9 @@ type elemBase struct {
 
 	// Mutable states.
 	cachedSourceSets []map[uint32]*bitset.BitSet // nolint: structcheck
-	// a cache of the flush metrics that don't require grabbing a lock to access.
+	// a cache of the metrics that don't require grabbing a lock to access.
```

Comment:
This comment is a little misleading since writeMetricsCache does require a lock.

Reply:
Nvm I see it now.